### PR TITLE
[LO-970] Test suite passing

### PR DESCRIPTION
**Pull request by thermistor**
There are still some non-deterministic features specs, however all of them pass individually when examined.

### CODE REVIEW

**Comment by rbarreca** [comment]
@thermistor next time you're at it, could you maybe add the folders with remaining failing specs and we could put our names next to each folder as we're working on it so we don't duplicate effort?

File: circle.yml [hunk 1]
```diff
@@ -5,11 +5,9 @@ machine:
     USE_STRIPE_STANDALONE_ACCOUNTS: 'true'
 dependencies:
   pre:
+    - gem update bundler
     - sudo apt-get install libjpeg62; sudo apt-get install ghostscript
     - sudo apt-get update; sudo apt-get install libicu52
-    - curl --output /home/ubuntu/bin/phantomjs-2.0.1-linux-x86_64-dynamic https://s3.amazonaws.com/circle-support-bucket/phantomjs/phantomjs-2.0.1-linux-x86_64-dynamic
-    - chmod a+x /home/ubuntu/bin/phantomjs-2.0.1-linux-x86_64-dynamic
-    - sudo ln -s --force /home/ubuntu/bin/phantomjs-2.0.1-linux-x86_64-dynamic /usr/local/bin/phantomjs
```
Comment (ivar) [inline circle.yml]: does circle work now or ?

File: lib/orders/unit_price_logic.rb [hunk 2]
```diff
@@ -2,9 +2,9 @@ module Orders
   class UnitPriceLogic
     class << self
       def prices(product, market, organization, time)
-        price = Price.for_product_and_market_and_org_at_time(product, market, organization, time).visible.order("min_quantity, organization_id")
+        price = Price.for_product_and_market_and_org_at_time(product, market, organization, time).order("min_quantity, organization_id")
         if price.empty?
-          price = Price.for_product_and_market_and_org_at_time(product, market, organization, Time.current).visible.order(:organization_id, :min_quantity)
+          price = Price.for_product_and_market_and_org_at_time(product, market, organization, Time.current).order(:organization_id, :min_quantity)
```
Comment (ivar) [inline lib/orders/unit_price_logic.rb]: inconsistent typing of 'order' argument..

File: circle.yml (hunk 1 above)
Comment (rbarreca) [reply circle.yml]: https://circleci.com/gh/LocalOrbit/localorbit/6216 says timeout and has some failures...

File: spec/features/market_manager/signing_in_spec.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@
   let!(:market2) { create(:market) }
   let!(:market1) { create(:market, managers: [user]) }
 
-  let!(:org) { create(:organization, markets: [market2], users: [user]) }
+  let!(:org) { create(:organization, org_type: 'M', markets: [market2], users: [user]) }
```
Comment (ivar) [inline spec/features/market_manager/signing_in_spec.rb]: type literals have been replaced with constants..

File: spec/support/capybara.rb [hunk 4]
```diff
@@ -28,3 +28,38 @@
...
+#             reload if Capybara.automatic_reload
+#             retry
+#           ensure
+#             session.synchronized = false
+#           end
+#         end
+#       end
+#     end
+#   end
+# end
```
Comment (ivar) [inline spec/support/capybara.rb]: probably best not to include commented code like this

**Comment by ivar** [comment]
Too much to parse all in one go.. I skipped the changes to the cassettes and by the second half I was spot checking instead of going through each one. Just some minor comments, none of which are showstoppers. There are some deleted specs where it'd be good to understand why they were deleted (irrelevant, can't make it work, not applicable, etc...)

File: circle.yml (hunk 1 above)
Comment (thermistor) [reply circle.yml]: Don't worry about Circle, it's been timing out for a long time. I'm working  on Circle 2.0

File: lib/orders/unit_price_logic.rb (hunk 2 above)
Comment (thermistor) [reply lib/orders/unit_price_logic.rb]: @ivar not sure what you mean ...

**Comment by thermistor** [comment]
@ivar was there a deleted spec that you couldn't see a reason in the appropriate git commit message? b367f3363 was one that can be hard to find.

File: app/assets/javascripts/products.js.coffee [hunk 5]
```diff
@@ -262,13 +262,15 @@ $ ->
...
-        organization_id: $("select[name='product[organization_id]'] option:selected").val()
-      }
-      error: (jqXHR, textStatus, errorThrown) ->
-        console.log("AJAX Error: #{textStatus}")
-      success: (data, textStatus, jqXHR) ->
-        console.log("Dynamic country select OK!")
\ No newline at end of file
+    organizationId = $("select[name='product[organization_id]'] option:selected").val();
+    if organizationId
+      $.ajax 'update_supplier_products',
```
[the remark this answers is not in the document]
Comment (thermistor) [reply app/assets/javascripts/products.js.coffee]: It fixes a user-facing bug, AND it makes the spec pass. Yay.

[the remark this answers is not in the document]
Comment (thermistor) [reply app/assets/javascripts/products.js.coffee]: I've added to the manual test plan.

**Comment by ivar** [comment]
looks good! :shipit:

**Pull request merged**